api/apps/api/src/modules/scenarios/scenarios.controller.ts

```diff
@@ -1091,4 +1098,34 @@ export class ScenariosController {
       throw new ForbiddenException();
     }
   }
+
+  @ApiOperation({
+    description: `Start Scenario Editing session.`,
+    summary: `Creates a Lock entity with related scenario
+     and user data to prevent other users from editing at the same time.`,
+  })
+  @Post(`:id/lock`)
+  async startScenarioEditingSession(
+    @Param('id', ParseUUIDPipe) id: string,
+    @Req() req: RequestWithAuthenticatedUser,
+  ): Promise<ScenarioLockResult> {
+    const result = await this.lockService.acquireLock(id, req.user.id);
+
+    if (isLeft(result)) {
+      switch (result.left) {
+        case forbiddenError:
+          throw new ForbiddenException();
+        case lockedScenario:
+          throw new BadRequestException(
+            `Scenario ${id} is already being edited`,
```

Comment:
```suggestion
            `Scenario ${id} is already being edited.`,
```

Extremely minor and cosmetic only. I don't think we have a guideline on this nor clear unique convention across existing code, but I'd personally lean on using a full stop at the end of messages in general.